var/spack/repos/builtin/packages/axom/package.py

```diff
@@ -65,11 +62,11 @@ def submodules(self):
     # Variants
     # -----------------------------------------------------------------------
     variant("shared", default=True, description="Enable build of shared libraries")
-    variant("debug", default=False, description="Build debug instead of optimized version")
 
     variant("examples", default=True, description="Build examples")
     variant("tools", default=True, description="Build tools")
 
+    # Hard requirement after Axom 0.6.1
     variant("cpp14", default=True, description="Build with C++14 support")
```

Comment:
We should make this variant only appear for `axom@:0.6.0`

`variant("cpp14", default=True, when="@:0.6.0", ...)`

doesn't need to be part of this PR